### PR TITLE
Volume controls in Game Options window

### PR DESCRIPTION
**Pull request by nielsmh**
## Motivation / Problem
You can't control sound and music volume from the main menu.
Several new players complain about being blasted by sound when they start the game for the first time.

## Description
Add volume controls to the Game Options window. The controls remain in the Jukebox window too, the controls are synced between both windows.

Also change the default volume of sound and music from "max" to something less. (The default of Transport Tycoon wasn't max.)

https://user-images.githubusercontent.com/1062071/113504681-e86c0580-9539-11eb-815e-c10773dcae77.mp4

## Limitations
Whether to change the default volume settings is probably up to discussion.
Arguably the volume slider control should be implemented as a proper widget instead of the slightly hacky pair of functions done here.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')

### CODE REVIEW

**Comment by LordAro** [comment]
Ordinarily I'm not sure this would be backported, but...

File: src/settings_gui.cpp [hunk 1]
```diff
@@ -320,6 +321,8 @@ struct GameOptionsWindow : Window {
 
 	void DrawWidget(const Rect &r, int widget) const override
 	{
+		extern void DrawVolumeSliderWidget(Rect r, byte value); // music_gui.cpp
```
Comment (LordAro) [inline src/settings_gui.cpp]: 2 separate externs here now, perhaps can include some header instead?

File: src/settings_gui.cpp [hunk 2]
```diff
@@ -441,6 +462,21 @@ struct GameOptionsWindow : Window {
 				this->SetDirty();
 				break;
 
+			case WID_GO_BASE_SFX_VOLUME:
+			case WID_GO_BASE_MUSIC_VOLUME: {
+				extern bool ClickVolumeSliderWidget(Rect r, Point pt, byte & value);
```
Comment (LordAro) [inline src/settings_gui.cpp]:
```suggestion
				extern bool ClickVolumeSliderWidget(Rect r, Point pt, byte &value);
```
(assuming no include change)

**Comment by michicc** [comment]
Might not strictly be in-scope, but I'd think at least a (combined) start/stop button next to the music volume would make sense, too.

**Comment by nielsmh** [comment]
> 
> 
> Might not strictly be in-scope, but I'd think at least a (combined) start/stop button next to the music volume would make sense, too.

Well yes, and even then you can always just change to NoMusic set instead. But that might not be very intuitive.

**Review by michicc — APPROVED** [review]

**Pull request merged**